src/main/java/com/salesforce/storm/spout/dynamic/DynamicSpout.java

```diff
@@ -491,6 +511,16 @@ String getOutputStreamId() {
         return outputStreamId;
     }
 
+    /**
+     * @return The stream that tuples that have permanently failed will be emitted out.
+     */
+    String getPermanentlyFailedOutputStreamId() {
```

Comment:
Assumption here is that we're just taking the standard output stream and tacking on "_failed".

Other option is to allow explicitly defining this value.

Or we could just hard code it to something like "permanently_failed"

Or some other option?

Reply:
This seems like it should probably be driven by a configuration value with a decent default.

Reply:
added SpoutConfig.PERMANENTLY_FAILED_OUTPUT_STREAM_ID => "spout.permanently_failed_output_stream_id" with a default value of "failed"